src/Agent/NewRelic/Agent/Extensions/Providers/Wrapper/Log4NetLogging/Log4netWrapper.cs

```diff
@@ -57,7 +64,27 @@ private void RecordLogMessage(object logEvent, Type logEventType, IAgent agent)
             // Older versions of log4net only allow access to a timestamp in local time
             var getTimestampFunc = _getTimestamp ??= VisibilityBypasser.Instance.GeneratePropertyAccessor<DateTime>(logEventType, "TimeStamp");
 
-            var getLogExceptionFunc = _getLogException ??= VisibilityBypasser.Instance.GeneratePropertyAccessor<Exception>(logEventType, "ExceptionObject");
+            Func<object, Exception> getLogExceptionFunc;
+
+            try
+            {
+                getLogExceptionFunc = _getLogException ??= VisibilityBypasser.Instance.GeneratePropertyAccessor<Exception>(logEventType, "ExceptionObject");
+            }
+            catch
```

Comment:
Would it be better if we used reflection to see if that Property (or field in the other block) exists instead of relying on this sequence of try-catch blocks?

Reply:
That would probably be a little cleaner, but would mean some refactoring in the VisibilityBypasser. Maybe something for the backlog? This will only throw once if you're running Sitecore, and then the function will be cached, so I don't think there's much overhead (unless I'm missing something).

Reply:
Overall it would be a 1-time cost so the difference probably would not be noticeable. In the past we didn't perform that check in the visibility bypasser. There is at least one other spot in instrumentation where we needed to do something similar, and took the approach of using reflection to check for the existence of something before attempting to setup the visibility bypasser. I don't have a strong reason to prefer one approach over the other. However, it might make sense like you're suggesting to have something in the backlog so that it's easier to use a consistent pattern (possibly in the VisibilityBypasser or something similar).

Reply:
Yeah I think a common pattern would be a good idea. I've created a ticket to come back to it: https://github.com/newrelic/newrelic-dotnet-agent/issues/1797